embassy-nrf/src/pwm.rs

```diff
@@ -1,16 +1,296 @@
 #![macro_use]
 
-use core::cell::UnsafeCell;
 use core::marker::PhantomData;
 use core::sync::atomic::{compiler_fence, Ordering};
 use embassy::util::Unborrow;
 use embassy_hal_common::unborrow;
 
 use crate::gpio::sealed::Pin as _;
-use crate::gpio::OptionalPin as GpioOptionalPin;
+use crate::gpio::{AnyPin, OptionalPin as GpioOptionalPin};
 use crate::interrupt::Interrupt;
 use crate::pac;
+use crate::util::slice_in_ram_or;
 
+/// Interface to the PWM peripheral
+pub struct SimplePwm<'d, T: Instance> {
+    phantom: PhantomData<&'d mut T>,
+    duty: [u16; 4],
+    ch0: Option<AnyPin>,
+    ch1: Option<AnyPin>,
+    ch2: Option<AnyPin>,
+    ch3: Option<AnyPin>,
+}
+
+pub struct SequencePwm<'d, T: Instance> {
+    phantom: PhantomData<&'d mut T>,
+    ch0: Option<AnyPin>,
+    ch1: Option<AnyPin>,
+    ch2: Option<AnyPin>,
+    ch3: Option<AnyPin>,
+}
+
```

Comment:
Is it possible to avoid the name, "Simple". Simple is definitely a matter of personal perspective and we don't have it anywhere else. May I suggest just `Pwm` and `ExtendedPwm`? Also, I feel that both of these structs should be documented. Doing so will help yield the name.

Reply:
Or, `Pwm` with an optional config and sequence.

Or, `DutyPwm` and `SequencePwm`.

Reply:
How about `SingleDutyPwm`? I think this highlights the difference between it and sequencing.

Accordingly, can we change the doc comment to:

```
/// SingleDutyPwm sets a single duty cycle on up to four channels.
```

Along the above lines, `DutySequencePwm` may be more descriptive than `SequencePwm`, with its description changed to:

```
/// DutySequencePwm specifies a sequence of duty cycles on up to four channels
/// and repeat that sequence n times.
```